internal/containerupdate/exec_updater.go

```diff
@@ -50,9 +52,18 @@ func (cu *ExecUpdater) UpdateContainer(ctx context.Context, cInfo liveupdates.Co
 	// copy files to container
 	buf := bytes.NewBuffer(nil)
 	tarWriter := io.MultiWriter(w, buf)
+	tarCmd := tarCmd()
 	err := cu.kCli.Exec(ctx, cInfo.PodID, cInfo.ContainerName, cInfo.Namespace,
-		tarArgv(), archiveToCopy, tarWriter, tarWriter)
+		tarCmd.Argv, archiveToCopy, tarWriter, tarWriter)
 	if err != nil {
+		if exitCodeErr, ok := build.WrapCodeExitError(err, tarCmd); ok {
+			switch exitCodeErr.ExitCode {
+			case TarExitCodePermissionDenied:
+				return permissionDeniedErr(err)
+			case GenericExitCodeCannotExec:
+				return cannotExecErr(err)
```

Comment:
Is it worth extracting this switch to a new function? Or moving to build.WrapCodeExitError()?

Reply:
(see top-level comment - I tried to refactor for this + some other improvements)